saleor/payment/migrations/0042_add_calculation_transaction_events.py

```diff
@@ -0,0 +1,126 @@
+from decimal import Decimal
+
+from django.db import migrations
+from django.db.models import F, Q, OuterRef, Subquery, Func, Exists
+
+
+def create_transaction_events(apps, schema_editor):
+    TransactionItem = apps.get_model("payment", "TransactionItem")
+    TransactionEvent = apps.get_model("payment", "TransactionEvent")
+    create_event_for_authorized(TransactionItem, TransactionEvent)
+    create_event_for_charged(TransactionItem, TransactionEvent)
+    create_event_for_refunded(TransactionItem, TransactionEvent)
+    create_event_for_canceled(TransactionItem, TransactionEvent)
+
+
+def _create_transaction_events(TransactionEvent, transactions_qs, type):
+    transaction_events = []
+    for pk, amount, amount_sum, currency in transactions_qs:
```

Comment:
Actually this topic was already raised:
https://github.com/saleor/saleor/pull/11937#discussion_r1105443731

But If there is any concern it will not be to much of effort to make it into bulk update.